moped-editor/src/views/projects/projectView/ProjectSummary/SubprojectsTable.js

```diff
@@ -1,38 +1,131 @@
-} from "@material-table/core";
+  DataGridPro,
+  GridRowModes,
+  GridActionsCellItem,
+} from "@mui/x-data-grid-pro";
+import { v4 as uuidv4 } from "uuid";
+
+import dataGridProStyleOverrides from "src/styles/dataGridProStylesOverrides";
+import SubprojectsToolbar from "./SubprojectsToolbar";
 import ApolloErrorHandler from "../../../../components/ApolloErrorHandler";
 import ProjectStatusBadge from "../../projectView/ProjectStatusBadge";
-import RenderFieldLink from "../../../../components/RenderFieldLink";
+import SubprojectLookupComponent from "./SubprojectLookupComponent";
+import DeleteConfirmationModal from "../DeleteConfirmationModal";
+import RenderFieldLink from "src/components/RenderFieldLink";
 
 import {
   SUBPROJECT_QUERY,
   UPDATE_PROJECT_SUBPROJECT,
   DELETE_PROJECT_SUBPROJECT,
 } from "../../../../queries/subprojects";
-import typography from "../../../../theme/typography";
 
-const SubprojectsTable = ({ projectId = null, refetchSummaryData }) => {
-  const addActionRef = React.useRef();
+/** Hook that provides memoized column settings */
+const useColumns = ({
+  data,
+  rowModesModel,
+  handleDeleteOpen,
+  handleSaveClick,
+  handleCancelClick,
+}) =>
+  useMemo(() => {
+    return [
+      {
+        headerName: "ID",
+        field: "project_id",
+        editable: false,
+        width: 75,
+      },
+      {
+        headerName: "Full name",
+        field: "project_name_full",
+        editable: true,
+        width: 250,
+        renderCell: ({ row }) => (
+          <RenderFieldLink
+            projectId={row?.project_id}
+            value={row?.project_name_full}
+          />
+        ),
+        renderEditCell: (props) => (
+          <SubprojectLookupComponent {...props} data={data} />
+        ),
+      },
+      {
+        headerName: "Status",
+        field: "moped_proj_phases",
+        editable: false,
+        width: 200,
+        // valueGetter allows us to derive from a nested field which will be used for sorting/filtering
+        valueGetter: (value) =>
+          value?.[0]?.moped_phase?.phase_name || "Unknown",
+        renderCell: ({ row }) =>
+          // only render a badge once we exit edit mode and there is a phase
+          row.moped_proj_phases ? (
+            <ProjectStatusBadge
+              phaseName={row?.moped_proj_phases?.[0]?.moped_phase?.phase_name}
+              phaseKey={row?.moped_proj_phases?.[0]?.moped_phase?.phase_key}
+              condensed
+            />
+          ) : (
+            <div />
+          ),
+      },
+      {
+        headerName: "",
+        field: "edit",
+        hideable: false,
+        filterable: false,
+        sortable: false,
+        editable: false,
+        type: "actions",
+        getActions: ({ id }) => {
+          const isInEditMode = rowModesModel[id]?.mode === GridRowModes.Edit;
+          if (isInEditMode) {
+            return [
+              <GridActionsCellItem
+                icon={<CheckIcon sx={{ fontSize: "24px" }} />}
```

Comment:
based on some feedback, I made a variable for this 
https://github.com/cityofaustin/atd-moped/blob/main/moped-editor/src/utils/dataGridHelpers.js#L1

Reply:
cool i just pulled that in!